esp-hal/src/dma/mod.rs

```diff
@@ -48,6 +48,137 @@
 
 use core::{fmt::Debug, marker::PhantomData, ptr::addr_of_mut, sync::atomic::compiler_fence};
 
+trait Word {}
+
+impl Word for u8 {}
+impl Word for u16 {}
+impl Word for u32 {}
+impl Word for i8 {}
+impl Word for i16 {}
+impl Word for i32 {}
+
+impl<W, const S: usize> crate::private::Sealed for [W; S] where W: Word {}
+
+impl<W, const S: usize> crate::private::Sealed for &[W; S] where W: Word {}
+
+impl<W> crate::private::Sealed for &[W] where W: Word {}
+
+impl<W> crate::private::Sealed for &mut [W] where W: Word {}
+
+/// Trait for buffers that can be given to DMA for reading.
+pub trait ReadBuffer: crate::private::Sealed {
```

Comment:
I wonder if we still need the `ReadBuffer` and `WriteBuffer` traits, could we just take `&[W]` and `&mut [W]` in the signature instead, `where W: Word`?

Reply:
I thought of s.th. like that but decided for the traits for (maybe) future things like specialized buffers with other alignment requirements etc. - no sure if that is really worth it but on the other hand it's not too much bloat